apollo/plugin.py

```diff
@@ -0,0 +1,175 @@
+from datetime import datetime
+from dataclasses import dataclass
+import httpx
+from typing import Optional, List
+
+from lutraai.augmented_request_client import AugmentedTransport
+
+
+@dataclass
+class ApolloEmploymentHistory:
+    organization_name: Optional[str]
+    start_date: Optional[datetime]
+    end_date: Optional[datetime]
+    title: Optional[str]
+
+
+@dataclass
+class ApolloFundingEvent:
+    date: datetime
+    news_url: Optional[str]
+    type: str
+    investors: List[str]
+    amount: str
+    currency: str
+
+
+@dataclass
+class ApolloOrganization:
+    name: str
+    website_url: Optional[str]
+    blog_url: Optional[str]
+    angellist_url: Optional[str]
+    twitter_url: Optional[str]
+    facebook_url: Optional[str]
+    phone: Optional[str]
+    founding_year: Optional[int]
+    industries: List[str]
+    keywords: List[str]
+    estimated_num_employees: Optional[int]
+    street_address: Optional[str]
+    city: Optional[str]
+    state: Optional[str]
+    country: Optional[str]
+    seo_description: Optional[str]
+    annual_revenue: Optional[float]
+    funding_events: List[ApolloFundingEvent]
+
+
+@dataclass
+class ApolloPersonProfile:
+    first_name: str
+    last_name: str
+    full_name: str
+    linkedin_url: Optional[str]
+    twitter_url: Optional[str]
+    github_url: Optional[str]
+    facebook_url: Optional[str]
+    headline: Optional[str]
+    email: Optional[str]
+    state: Optional[str]
+    city: Optional[str]
+    country: Optional[str]
+    employment_history: List[ApolloEmploymentHistory]
+    organization: ApolloOrganization
+    departments: List[str]
+    subdepartments: List[str]
+    seniority: Optional[str]
+
+
+def _parse_apollo_date(date_str: Optional[str]) -> Optional[datetime]:
+    if date_str:
+        return datetime.strptime(date_str, "%Y-%m-%d")
```

Comment:
might want to try-catch around this since strptime  might raise an error -- and in those cases, you want to return None 

(future follow up)